dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/HTMLPageAssetAPIImplTest.java

```diff
@@ -0,0 +1,107 @@
+import com.dotmarketing.exception.WebAssetException;
+import com.dotmarketing.factories.PublishFactory;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.UUIDGenerator;
+import java.util.Collections;
+import java.util.Date;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class HTMLPageAssetAPIImplTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Given scenario:
+     * a new content in a second language (not default) belonging to a htmlpage
+     *
+     * Expected result:
+     * getHTML method should respond with the proper rendered content on the belonging htmlpage
+     */
+
+    @Test
+    public void test_getHTML_GivenSpanishOnlyContent_shouldReturnHTML()
+            throws DotSecurityException, DotDataException, WebAssetException {
+        final Host site = new SiteDataGen().nextPersisted();
+        final Language language = new LanguageDataGen().nextPersisted();
+
+        final ContentType blogType = TestDataUtils.getBlogLikeContentType(site);
+
+        // create detail page
+        final Container container = new ContainerDataGen()
+                .withContentType(blogType, "$body")
+                .nextPersisted();
+
+        ContainerDataGen.publish(container);
+
+        final String uuid = UUIDGenerator.generateUuid();
+
+        final Template template = new TemplateDataGen().withContainer(container.getIdentifier(), uuid)
+                .nextPersisted();
+
+        TemplateDataGen.publish(template);
+
+        final HTMLPageAsset htmlPage = new HTMLPageDataGen(site, template)
+                .pageURL("blog-detail")
+                .title("blog-detail")
+                .nextPersisted();
+
+        HTMLPageDataGen.publish(htmlPage);
+
+        // create URL-Mapped content
+        final Contentlet urlMappedContent = new ContentletDataGen(blogType.id())
+                .languageId(language.getId())
+                .setProperty("body", "myBody")
+                .nextPersisted();
+
+        ContentletDataGen.publish(urlMappedContent);
+
+        final MultiTree multiTree = new MultiTree(htmlPage.getIdentifier(),
+                container.getIdentifier(),
+                urlMappedContent.getIdentifier(),
+                getDotParserContainerUUID(uuid), 0);
+
+        APILocator.getMultiTreeAPI().saveMultiTree(multiTree);
+
+        String html = APILocator.getHTMLPageAssetAPI().getHTML(htmlPage.getURI(), site, true,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'html' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5189840)